blkmaker.c

```diff
@@ -322,21 +322,21 @@ bool _blkmk_calculate_witness_mrklroot(blktemplate_t * const tmpl, libblkmaker_h
 		if (txn->txid && memcmp(txn->hash_, txn->txid, sizeof(*txn->txid))) {
 			*witness_needed = true;
 		}
-		memcpy(&hashes[i + 1], txn->hash_, sizeof(*hashes));
+		memcpy(&hashes[i + 1], txn->hash_, sizeof(libblkmaker_hash_t));
```

Comment:
I was nervous about the VLA interaction here so i made it explicit while testing to rule that out. I left it in because it works for sure this way and it's more readable imo. I can remove if you'd prefer.